stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Channel.kt

```diff
@@ -90,6 +88,16 @@ public data class Channel(
     override var extraData: MutableMap<String, Any> = mutableMapOf(),
 ) : CustomObject, ComparableFieldProvider {
 
+    /**
+     * The channel id in the format messaging:123.
+     */
+    val cid: String
+        get() = if (id.isEmpty() || type.isEmpty()) {
+            ""
+        } else {
+            "$type:$id"
+        }
```

Comment:
What is the problem returning always `$type:$id`? :thinking:

Reply:
1. I think it's better to simply return `""` when either type or id is empty as we are not able to create valid cid
2. In Compose we have this code:
```val channel = selectedChannel ?: Channel()
AnimatedVisibility(
     visible = channel.cid.isNotEmpty(), // not empty here, cid is ":" here
     enter = fadeIn(),
     exit = fadeOut(animationSpec = tween(durationMillis = AnimationConstants.DefaultDurationMillis / 2))
) {
}```
for showing the channel's options.

Reply:
Yep, for convenience not to deal with nullable strings we do:

```
val channel = selectedChannel ?: Channel()
AnimatedVisibility(
      visible = channel.cid.isNotEmpty(),
     enter = fadeIn(),
      exit = fadeOut(animationSpec = tween(durationMillis = AnimationConstants.DefaultDurationMillis / 2))
)
``` 
And "$type:$id" will evaluate to ":" which breaks our UI.